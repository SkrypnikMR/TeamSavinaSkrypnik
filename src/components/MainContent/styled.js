import styled from 'styled-components';

export const StMainContent = styled.div`
    background: #696969;
    margin: 10px;
    height: 84vh;
    display: flex;
`