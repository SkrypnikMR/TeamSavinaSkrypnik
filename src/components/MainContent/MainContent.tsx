import React from 'react';
import RoomsContainer from '../RoomsContainer';
import { StMainContent } from './styled';
import GameContent from '../GameContent';

const MainContent = () => {
    return (
        <StMainContent>
            <RoomsContainer/>
            <GameContent />
        </StMainContent>
    );
};

export default MainContent;
