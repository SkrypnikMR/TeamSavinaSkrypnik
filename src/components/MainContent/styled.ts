import styled from 'styled-components';

import { midGrey } from '../UI/baseLayout';

export const StMainContent = styled.div`
    background: ${midGrey};
    margin: 10px;
    height: 85vh;
    display: flex;
`;
