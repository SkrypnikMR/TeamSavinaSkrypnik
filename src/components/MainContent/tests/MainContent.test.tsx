import React from 'react';
import MainContent from '../MainContent';
import { shallowSmart, mountSmart } from '../../../../__tests__/testHelper';


describe('MainContent', () => {
    it('Should match snapshot', () => {
        const component = shallowSmart(<MainContent />);
        expect(component.html()).toMatchSnapshot();
    });
    it('should render StRoomsContainer', () => {
        const component = mountSmart(<MainContent/>);
        expect(component.find('styled__StMainContent')).toHaveLength(1);
    });
    it('should render SingleRoom', () => {
        const component = mountSmart(<MainContent/>);
        expect(component.find('RoomsContainer')).toHaveLength(1);
    });
    it('should render AddRoom', () => {
        const component = mountSmart(<MainContent/>);
        expect(component.find('GameContent')).toHaveLength(1);
    });
});
