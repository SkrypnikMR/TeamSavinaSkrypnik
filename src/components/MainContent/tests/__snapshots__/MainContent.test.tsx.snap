// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContent Should match snapshot 1`] = `"<div class=\\"styled__StMainContent-sc-1j4ztaa-0 liFTEK\\"><div class=\\"styled__StRoomsContainer-sc-2jyga5-0 dzQjZf\\"><div class=\\"styled__StSingleRoom-sc-1b2p865-0 iXMywQ\\">Who create<button type=\\"button\\" width=\\"70px\\" class=\\"styled__StButton-sc-1y35zax-0 fkzlAn\\">Join</button></div><div class=\\"styled__StSingleRoom-sc-1b2p865-0 iXMywQ\\">Who create<button type=\\"button\\" width=\\"70px\\" class=\\"styled__StButton-sc-1y35zax-0 fkzlAn\\">Join</button></div><div class=\\"styled__StSingleRoom-sc-1b2p865-0 iXMywQ\\">Who create<button type=\\"button\\" width=\\"70px\\" class=\\"styled__StButton-sc-1y35zax-0 fkzlAn\\">Join</button></div><div class=\\"styled__StAddRoom-sc-5w938c-0 kQWGyo\\">+</div></div><div class=\\"styled__StGameContent-sc-zisfga-0 jHdPPO\\">Checkers<div class=\\"styled__StGameZone-sc-1e1y6pd-0 gFukiy\\">your turn<div class=\\"styled__StChessPlate-sc-cyrsg4-0 kHdeUV\\"><div id=\\"64\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">64</div><div id=\\"63\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">63</div><div id=\\"62\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">62</div><div id=\\"61\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">61</div><div id=\\"60\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">60</div><div id=\\"59\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">59</div><div id=\\"58\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">58</div><div id=\\"57\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">57</div><div id=\\"56\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">56</div><div id=\\"55\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">55</div><div id=\\"54\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">54</div><div id=\\"53\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">53</div><div id=\\"52\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">52</div><div id=\\"51\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">51</div><div id=\\"50\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">50</div><div id=\\"49\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">49</div><div id=\\"48\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">48</div><div id=\\"47\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">47</div><div id=\\"46\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">46</div><div id=\\"45\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">45</div><div id=\\"44\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">44</div><div id=\\"43\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">43</div><div id=\\"42\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">42</div><div id=\\"41\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">41</div><div id=\\"40\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">40</div><div id=\\"39\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">39</div><div id=\\"38\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">38</div><div id=\\"37\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">37</div><div id=\\"36\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">36</div><div id=\\"35\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">35</div><div id=\\"34\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">34</div><div id=\\"33\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">33</div><div id=\\"32\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">32</div><div id=\\"31\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">31</div><div id=\\"30\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">30</div><div id=\\"29\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">29</div><div id=\\"28\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">28</div><div id=\\"27\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">27</div><div id=\\"26\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">26</div><div id=\\"25\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">25</div><div id=\\"24\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">24</div><div id=\\"23\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">23</div><div id=\\"22\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">22</div><div id=\\"21\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">21</div><div id=\\"20\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">20</div><div id=\\"19\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">19</div><div id=\\"18\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">18</div><div id=\\"17\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">17</div><div id=\\"16\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">16</div><div id=\\"15\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">15</div><div id=\\"14\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">14</div><div id=\\"13\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">13</div><div id=\\"12\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">12</div><div id=\\"11\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">11</div><div id=\\"10\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">10</div><div id=\\"9\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">9</div><div id=\\"8\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">8</div><div id=\\"7\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">7</div><div id=\\"6\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">6</div><div id=\\"5\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">5</div><div id=\\"4\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">4</div><div id=\\"3\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">3</div><div id=\\"2\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">2</div><div id=\\"1\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">1</div></div></div></div></div>"`;

exports[`RoomsContainer Should match snapshot 1`] = `"<div class=\\"styled__StMainContent-sc-1j4ztaa-0 liFTEK\\"><div class=\\"styled__StRoomsContainer-sc-2jyga5-0 dzQjZf\\"><div class=\\"styled__StSingleRoom-sc-1b2p865-0 iXMywQ\\">Who create<button type=\\"button\\" width=\\"70px\\" class=\\"styled__StButton-sc-1y35zax-0 fkzlAn\\">Join</button></div><div class=\\"styled__StSingleRoom-sc-1b2p865-0 iXMywQ\\">Who create<button type=\\"button\\" width=\\"70px\\" class=\\"styled__StButton-sc-1y35zax-0 fkzlAn\\">Join</button></div><div class=\\"styled__StSingleRoom-sc-1b2p865-0 iXMywQ\\">Who create<button type=\\"button\\" width=\\"70px\\" class=\\"styled__StButton-sc-1y35zax-0 fkzlAn\\">Join</button></div><div class=\\"styled__StAddRoom-sc-5w938c-0 kQWGyo\\">+</div></div><div class=\\"styled__StGameContent-sc-zisfga-0 jHdPPO\\">Checkers<div class=\\"styled__StGameZone-sc-1e1y6pd-0 gFukiy\\">your turn<div class=\\"styled__StChessPlate-sc-cyrsg4-0 kHdeUV\\"><div id=\\"64\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">64</div><div id=\\"63\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">63</div><div id=\\"62\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">62</div><div id=\\"61\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">61</div><div id=\\"60\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">60</div><div id=\\"59\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">59</div><div id=\\"58\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">58</div><div id=\\"57\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">57</div><div id=\\"56\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">56</div><div id=\\"55\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">55</div><div id=\\"54\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">54</div><div id=\\"53\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">53</div><div id=\\"52\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">52</div><div id=\\"51\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">51</div><div id=\\"50\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">50</div><div id=\\"49\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">49</div><div id=\\"48\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">48</div><div id=\\"47\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">47</div><div id=\\"46\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">46</div><div id=\\"45\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">45</div><div id=\\"44\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">44</div><div id=\\"43\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">43</div><div id=\\"42\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">42</div><div id=\\"41\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">41</div><div id=\\"40\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">40</div><div id=\\"39\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">39</div><div id=\\"38\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">38</div><div id=\\"37\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">37</div><div id=\\"36\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">36</div><div id=\\"35\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">35</div><div id=\\"34\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">34</div><div id=\\"33\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">33</div><div id=\\"32\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">32</div><div id=\\"31\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">31</div><div id=\\"30\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">30</div><div id=\\"29\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">29</div><div id=\\"28\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">28</div><div id=\\"27\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">27</div><div id=\\"26\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">26</div><div id=\\"25\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">25</div><div id=\\"24\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">24</div><div id=\\"23\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">23</div><div id=\\"22\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">22</div><div id=\\"21\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">21</div><div id=\\"20\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">20</div><div id=\\"19\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">19</div><div id=\\"18\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">18</div><div id=\\"17\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">17</div><div id=\\"16\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">16</div><div id=\\"15\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">15</div><div id=\\"14\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">14</div><div id=\\"13\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">13</div><div id=\\"12\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">12</div><div id=\\"11\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">11</div><div id=\\"10\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">10</div><div id=\\"9\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">9</div><div id=\\"8\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">8</div><div id=\\"7\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">7</div><div id=\\"6\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">6</div><div id=\\"5\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">5</div><div id=\\"4\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">4</div><div id=\\"3\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">3</div><div id=\\"2\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">2</div><div id=\\"1\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1k5unaq-0 bIotSN\\">1</div></div></div></div></div>"`;
